src/main/java/com/tysng/duke/ui/Command.java

```diff
@@ -0,0 +1,145 @@
+package com.tysng.duke.ui;
+
+import com.tysng.duke.domain.Deadline;
+import com.tysng.duke.domain.Event;
+import com.tysng.duke.domain.Task;
+import com.tysng.duke.domain.Todo;
+import com.tysng.duke.exception.CommandException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class parses raw user input text into a Command object that can be process by Duke.
+ */
+public class Command {
+    private CommandType type;
```

Comment:
IMO subclassing `Command` might be a cleaner approach than using enum and switch statements. It would avoid code duplication and cyclomatic complexity.

Reply:
Will consider. Thanks for the suggestion!